### ISSUE TITLE
.editorconfig file says to remove trailing whitespace

### ISSUE THREAD

**Issue by tomnelson** [jgrapht/jgrapht#1075]
Your .editorconfig file says to trim trailing whitespace:
```
trim_trailing_whitespace = true
```
Some IDEs (like IntelliJ) will honor that and introduce whitespace changes to your files where you have an extra space after an empty line in javadoc (you have asterisk plus space instead of only asterisk).

My first PR introduced those whitespace changes and you asked me to fix them. I see other PRs that also introduce changed lines where they trimmed the whitespace, for example: https://github.com/jgrapht/jgrapht/pull/1021/files

**Comment by jkinable** [comment]
Before releasing a new version of JGraphT we always run a code formatter to ensure that the source code looks pretty and consistent. As a consequence, there's no need for contributors to run their own formatting: we'd like our contributors to focus on the fun stuff, namely coding :). Since we run the formatting on our end, we don't have to bug contributors because they missed a white space, a line of code exceeds the max number of characters, or even worse, their curly brace is not on a new line!
Finally, the fewer files you change, the less likely it is that we end up with a merge conflict because 2 contributors edited the same file in two different PRs, and one PR got merged before the other. Nobody enjoys fixing merge requests.

When reviewing code, it's important that the reviewer quickly sees the relevant changes in a PR, i.e. not the cosmetic changes such as a new line or a removed space. Especially for large contributions, code reviewing becomes a lot harder if the important changes are somewhere in between a pile of cosmetic changes. Consequently, we ask our contributors not to format code outside of their own newly contributed code.

Unfortunately, some PRs with code formatting changes slip through the review process.
Long story short: don't bother with code formatting. Just make sure that your own, newly contributed code, is of good quality, and is readable.

**Comment by tomnelson** [comment]
I agree with all of the above. See https://github.com/jgrapht/jgrapht/pull/1072 for your request on my PR.

Its a little tricky to force IntelliJ to ignore your setting to remove trailing whitespace. I just went thru an issue with the IDEA team and they finally figured it out, that it was your .editorconfig file that was being honored on every save.

Mostly, I wanted to tell you why you are seeing those superfluous changes in pull requests, not just mine.

**Comment by tomnelson** [comment]
Also, FWIW, if I check out your jgrapht-1.5.1 tag, I see trailing spaces after * in places like this:
```
 * </pre>
 * 
 * <p>
 * In case the graph is weighted then the importer also reads edge weights. Otherwise the default
 * edge weight is returned. The importer also supports reading additional string attributes such as
 * label or custom user attributes.
 * 
 * <p>
```
Your formatter you run before release does not seem to be consistent with your .editorconfig file.

**Comment by jsichi** [comment]
@jkinable maybe we should comment out the line in the .editorconfig for now unless the autoformatting can be made 100% consistent with the rule?

**Comment by d-michail** [comment]
Code formatting is important only when submitting a PR, to be consistent with the code that already exists in the repository. Otherwise the `diff` contains formatting changes, and people cannot focus fast at the relevant changes.

@tomnelson In your particular PR this was not very important, as it was really small. But in bigger PRs this is very important.

**Comment by jkinable** [comment]
> @jkinable maybe we should comment out the line in the .editorconfig for now unless the autoformatting can be made 100% consistent with the rule?

I agree, let's disable this rule, and let the eclipse formatter handle it.
